library/core/src/main/java/com/google/android/exoplayer2/trackselection/DefaultTrackSelector.java

```diff
@@ -2736,7 +2753,17 @@ public int compareTo(TextTrackScore other) {
       if (this.hasPreferredIsForcedFlag != other.hasPreferredIsForcedFlag) {
         return this.hasPreferredIsForcedFlag ? 1 : -1;
       }
-      return compareInts(this.selectedAudioLanguageScore, other.selectedAudioLanguageScore);
+      if (this.selectedAudioLanguageScore != other.selectedAudioLanguageScore) {
+        return compareInts(this.selectedAudioLanguageScore, other.selectedAudioLanguageScore);
+      }
+
+      // If no role flag preference is provided in the parameters, prefer tracks
+      // without C.ROLE_FLAG_CAPTION or C.ROLE_FLAG_DESCRIBES_MUSIC_AND_SOUND
+      // No specification means that non-captioning text tracks are preferred.
+      if (!this.hasPreferredRoleFlags) {
+        return compareInts(other.hasCaptionRoleFlags, this.hasCaptionRoleFlags);
+      }
+      return compareInts(this.preferredRoleFlagsScore, other.preferredRoleFlagsScore);
```

Comment:
I actually meant that this comparison stays where it was (after the language score), but never mind, I just move it up again when merging.